src/main/java/converter/classes/Converter.java

```diff
@@ -1,23 +1,27 @@
 package converter.classes;
 
-import converter.exceptions.AmountOrExchangeException;
-import converter.interfaces.IConstantsConverter;
+import converter.exceptions.InvalidAmountOrExchangeRateException;
 import converter.interfaces.IRateProvider;
 
 import java.math.BigDecimal;
 import java.math.RoundingMode;
 
-public class Converter implements IConstantsConverter {
+public class Converter {
+	private final IRateProvider rateProvider;
```

Comment:
If you wanna use a different provider - just pass it into the constructor. Or use DI later